libraries/plugins/xfpga/metrics/metrics_max10.h

```diff
@@ -34,6 +34,7 @@
 
 
 #define DFL_MAX10_SYSFS_PATH                    "*dfl_*/*/**/*hwmon*/hwmon/hwmon*/"
```

Comment:
Hi Ananda, the patch keeps using the DFL_MAX10_SYSFS_PATH pattern for D5005 and N3000. We have not tested combinations of an N600xx card and D5005 in same server, to see if it could trigger the failure since we do not have running systems with those cards. But i think potentially the same problem could be provokede with combination of N600xx card and D5005 in same server. I expect the D5005 path from "*dfl_*" to "*hwmon*/hwmon/hwmon*" has a depth of 3 like "spi_master/spi0/spi0.0" 

/sys/class/fpga_region/region0/dfl-fme.0/dfl_dev.8/spi_master/spi0/spi0.0/n5014bmc-hwmon.5.auto/hwmon/hwmon7

but via the subsystem symlink there are other paths with depths 3:

/sys/class/fpga_region/region0/dfl-fme.0/dfl_dev.8/subsystem/devices/dfl_dev.11/n6000bmc-hwmon.2.auto/hwmon/hwmon4

Both paths satisfy the pattern and in our case, find_glob_path() finds the "n6000*" directory.

Reply:
Would it work if find_glob_path() does not follow symlinks at all? I think it is only /sys/class/fpga_region that have symlink to get forward in the hierarchy of sysfs files created by the drivers. This would be a change that requires verification for all card types. So we did not propose that.